### ISSUE TITLE
Support Function Apps

### ISSUE THREAD

**Issue by bbakermmc** [n3wt0n/AzureWebAppSSLManager#8]
The code currently doesnt work with Function Apps.

In the code AddCertificateToWebAppAsync
If webApp is null you should error, currently code breaks further down where you reference the the Region etc.

You will probably need to do this:  Then make a new DTO for the fields you need, you could split this if you wanted as a "Type" in the JSON
```c#
var funApp = _azure.AppServices.FunctionApps.ListByResourceGroup(_webAppResGroup).SingleOrDefault(w => w.Name == _webAppName.ToLower());
            var webApp = _azure.AppServices.WebApps.ListByResourceGroup(_webAppResGroup).SingleOrDefault(w => w.Name == _webAppName.ToLower());

There some .Where().SingleOrDefault() that can be replaced with just .SingleOrDefault
```

**Comment by bbakermmc** [comment]
You should also change the compares to something like this: .SingleOrDefault(w => w.Name.Equals(_webAppName, StringComparison.CurrentCultureIgnoreCase));   Your code was lowering one side but not the other.

**Comment by n3wt0n** [comment]
> You should also change the compares to something like this: .SingleOrDefault(w => w.Name.Equals(_webAppName, StringComparison.CurrentCultureIgnoreCase));   Your code was lowering one side but not the other.

Solved in #9.

> There some .Where().SingleOrDefault() that can be replaced with just .SingleOrDefault

Won't change this, the runtime code generation is the same but I find the '_.Where().SingleOrDefault()_' version more readable.
